samples/python/calibrate.py

```diff
@@ -30,31 +40,78 @@ def main():
+    args.setdefault('--h', 6)
+    args.setdefault('--t', 'chessboard')
+    args.setdefault('--square_size', 10)
+    args.setdefault('--marker_size', 5)
+    args.setdefault('--aruco_dict', 'DICT_4X4_50')
     args.setdefault('--threads', 4)
-    if not img_mask:
+
+    if not img_names:
         img_mask = '../data/left??.jpg'  # default
-    else:
-        img_mask = img_mask[0]
+        img_names = glob(img_mask)
 
-    img_names = glob(img_mask)
     debug_dir = args.get('--debug')
     if debug_dir and not os.path.isdir(debug_dir):
         os.mkdir(debug_dir)
-    square_size = float(args.get('--square_size'))
 
-    pattern_size = (9, 6)
-    pattern_points = np.zeros((np.prod(pattern_size), 3), np.float32)
-    pattern_points[:, :2] = np.indices(pattern_size).T.reshape(-1, 2)
-    pattern_points *= square_size
+    height = int(args.get('--h'))
+    width = int(args.get('--w'))
+    pattern_type = str(args.get('--t'))
+    square_size = float(args.get('--square_size'))
+    marker_size = float(args.get('--marker_size'))
+    aruco_dict_name = str(args.get('--aruco_dict'))
+
+    pattern_size = (height, width)
+    if pattern_type == 'chessboard':
+        pattern_points = np.zeros((np.prod(pattern_size), 3), np.float32)
+        pattern_points[:, :2] = np.indices(pattern_size).T.reshape(-1, 2)
+        pattern_points *= square_size
+    elif pattern_type == 'charucoboard':
+        pattern_points = np.zeros((np.prod((height-1, width-1)), 3), np.float32)
+        pattern_points[:, :2] = np.indices((height-1, width-1)).T.reshape(-1, 2)
+        pattern_points *= square_size
+    else:
+        print("unknown pattern")
+        return None
 
     obj_points = []
     img_points = []
     h, w = cv.imread(img_names[0], cv.IMREAD_GRAYSCALE).shape[:2]  # TODO: use imquery call to retrieve results
 
+    aruco_dicts = {
+        'DICT_4X4_50':cv.aruco.DICT_4X4_50,
+        'DICT_4X4_100':cv.aruco.DICT_4X4_100,
+        'DICT_4X4_250':cv.aruco.DICT_4X4_250,
+        'DICT_4X4_1000':cv.aruco.DICT_4X4_1000,
+        'DICT_5X5_50':cv.aruco.DICT_5X5_50,
+        'DICT_5X5_100':cv.aruco.DICT_5X5_100,
+        'DICT_5X5_250':cv.aruco.DICT_5X5_250,
+        'DICT_5X5_1000':cv.aruco.DICT_5X5_1000,
+        'DICT_6X6_50':cv.aruco.DICT_6X6_50,
+        'DICT_6X6_100':cv.aruco.DICT_6X6_100,
+        'DICT_6X6_250':cv.aruco.DICT_6X6_250,
+        'DICT_6X6_1000':cv.aruco.DICT_6X6_1000,
+        'DICT_7X7_50':cv.aruco.DICT_7X7_50,
+        'DICT_7X7_100':cv.aruco.DICT_7X7_100,
+        'DICT_7X7_250':cv.aruco.DICT_7X7_250,
+        'DICT_7X7_1000':cv.aruco.DICT_7X7_1000,
+        'DICT_ARUCO_ORIGINAL':cv.aruco.DICT_ARUCO_ORIGINAL,
+        'DICT_APRILTAG_16h5':cv.aruco.DICT_APRILTAG_16h5,
+        'DICT_APRILTAG_25h9':cv.aruco.DICT_APRILTAG_25h9,
+        'DICT_APRILTAG_36h10':cv.aruco.DICT_APRILTAG_36h10,
+        'DICT_APRILTAG_36h11':cv.aruco.DICT_APRILTAG_36h11
+    }
+
+    aruco_dict = cv.aruco.getPredefinedDictionary(aruco_dicts[aruco_dict_name])
```

Comment:
Please add check. that user defiles proper board name. Otherwise he/she gets some exception on dict access.